subsys/mgmt/mcumgr/Kconfig

```diff
@@ -7,7 +7,7 @@ menuconfig MCUMGR
 	depends on NET_BUF
 	depends on ZCBOR
 	depends on CRC
-	select BOOT_IMAGE_ACCESS_HOOKS if (SOC_NRF5340_CPUAPP_QKAA && MCUMGR_CMD_IMG_MGMT)
+	select BOOT_IMAGE_ACCESS_HOOKS if (SOC_NRF5340_CPUAPP_QKAA && MCUMGR_GRP_IMG)
```

Comment:
Can accept this change for upmerge as it replicates the existing code with new Kconfig name but this actually wrong, an nrf53 user only wants boot image hooks _if_ they are supporting network core firmware updates, if they are not, this should be disabled, as or right now it can't be disabled at all which is undesirable and wrong. @de-nordic

Reply:
I can see problem here, although I do not think there is not Kconfig magic to fix that. Only fix I can see here is to directly select the BOOT_IMAGE_ACCESS_HOOKS for configurations (samples, applications, test) that have net core as second image.

Reply:
A quick fix would be `IMPLY` as it allows people to disable it, probably better would be to tie this to the value of updateable image count and if PCD is enabled in the bootloader